bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoVacuumHandler.java

```diff
@@ -139,7 +201,8 @@ private boolean updateVacuumStatus(JsonObject statusData) {
         int fanLevel = statusData.get("fan_power").getAsInt();
         updateState(CHANNEL_FAN_POWER, new DecimalType(fanLevel));
         updateState(CHANNEL_FAN_CONTROL, new DecimalType(FanModeType.getType(fanLevel).getId()));
-        updateState(CHANNEL_IN_CLEANING, new DecimalType(statusData.get("in_cleaning").getAsBigDecimal()));
+        inCleaning = statusData.get("in_cleaning").getAsInt();
+        updateState(CHANNEL_IN_CLEANING, new DecimalType(fanLevel));
```

Comment:
fanLevel?

Reply:
:-) good catch... was doing bit trail & error testing.. seems I did not change this one back.